### ISSUE TITLE
aws_volume_attachment complains about "VolumeInUse" from imported resources

### ISSUE THREAD

**Issue by nicocesar** [hashicorp/terraform#8458]
### Terraform Version

Terraform v0.7.1
### Affected Resource(s)

aws_volume_attachment
### TF files

https://gist.github.com/nicocesar/378565e259cc5e5ae6076b571344f377
### Debug Output

https://gist.github.com/nicocesar/5e2e8f9c9290179b1fb0a61509cf30e8
### Expected Behavior

I've imported the volume and the instace using terraform import. But I don't know how to import the aws_volume_attachment. When I execute the terraform apply it tries to attach an _already attached_ volume. 

I just want to: a) do not report a missing attachment or b) know how to use terraform import to add this information too.
### Steps to Reproduce
1. terraform import aws_instance.shell i-afe8972f
2. terraform import aws_ebs_volume.shell_data vol-fbaa2524
3. terraform apply

**Comment by tdeheurles** [comment]
Hi, 
Our 2 issues seems close: https://github.com/hashicorp/terraform/issues/9910

Have you found a solution to this problem ? To me, we need to import the `aws_volume_attachment` but I don't know how to find it through aws.

**Comment by tdeheurles** [comment]
If it can help someone, I manually updated the aws_volume_attachment in the terraform state to match the new requirements. It seems that this aws_volume_attachment_id doesn't exist in amazon but only for terraform

**Comment by nicocesar** [comment]
Can you provide a diff of your "manual update"?

**Comment by tdeheurles** [comment]
Here is the part of the json for the `aws_volume_attachment` :
```json
"aws_volume_attachment.name": {
    "type": "aws_volume_attachment",
    "depends_on": [
        "aws_ebs_volume.name",
        "aws_instance.name"
    ],
    "primary": {
        "id": "vai-1501514762",
        "attributes": {
            "device_name": "xvdh",
            "id": "vai-1501514762",
            "instance_id": "i-0df597a608df3c6fb",
            "volume_id": "vol-d4398402"
        },
        "meta": {},
        "tainted": false
    },
    "deposed": [],
    "provider": ""
}
```

So I have just updated :
- `aws_volume_attachment.name.primary.attributes.instance_id`
- `aws_volume_attachment.name.primary.attributes.volume_id`
to match the new resources.

Note that these 2 resources were imported with `terraform import` before.

**Comment by tdna** [comment]
I have the same problem. Where does this id come from?

**Comment by foxsy** [comment]
This ID is generated when Terraform writes the attachment to the state file.  I had the same issue when I imported some resources and had to modify the state file by hand.  Since you can't use "terraform import" on a volume attachment your options are limited.  You can either force detach and let Terraform attach the volume, or you can manually update the state file and regenerate the "vai" ID.  To make this easier I pulled the relevant code that generates that ID and put it on github. https://github.com/foxsy/tfvolattid

**Comment by nicocesar** [comment]
Thanks for the relevant information, @foxsy  but why can't terraform just do that for you upon import? is there anything I'm missing here? by the time you do  terraform import aws_ebs_volume.shell_data vol-fbaa2524 (look in my example)  can't it ask you for a mount point or some extra info to make this happen without manual interventions... I have pending to migrate to terraform many clusters,  I don't want to do manually all this conversions

**Comment by stack72** [comment]
Closed via #11060
